### PR TITLE
JBPM-5095: default gate tests

### PR DESCRIPTION
**Pull request by jomarko**
Depends on #248

### CODE REVIEW

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/validation/BPMN2SyntaxChecker.java [hunk 1]
```diff
@@ -592,6 +594,12 @@ private void checkDataObject(DataObject dataObject) {
 		}
 	}
 
+	private void checkMandatoryDefaultGate(Gateway gateway, SequenceFlow defaultSequenceFlow) {
+		if(defaultSequenceFlow == null || (gateway.getOutgoing() != null && !gateway.getOutgoing().contains(defaultSequenceFlow))) {
+			addError(gateway, new ValidationSyntaxError(gateway, BPMN2_TYPE, SyntaxCheckerErrors.NOT_VALID_DEFAULT_GATE));
+		}
+	}
+
```
Comment (tsurdilo) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/validation/BPMN2SyntaxChecker.java]: Default Gate should always be optional. There is no hard requirement to have it set in the runtime as well as during process modeling. It is OK if the default gate is not specified so this check method is not valid.

What you can check is if a gateway does have a default specified if indeed it is one of the outgoing flows of it. That would be OK.

Comment (jomarko) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/validation/BPMN2SyntaxChecker.java]: Hello @tsurdilo. I agree, I will amend some changes into #248 and then I will close this PR. Thanks.

**Comment by jomarko** [comment]
Please do not merge yet.

**Comment by jomarko** [comment]
Our previous conversation has gone as I amended the PR but I think now it is OK. @tsurdilo what do you think?

**Comment by jomarko** [comment]
please retest this

**Comment by jomarko** [comment]
jenkins retest this

**Comment by tsurdilo** [comment]
@jomarko looks ok the only thing I would suggest to rename method "checkMandatoryDefaultGate" as we previously discussed the default gates are not "mandatory". checkDefaultGate would make more sense. WDYT?

**Comment by jomarko** [comment]
I agree, `checkDefaulGate` is more suitable name. PR ready for review.

**Comment by tsurdilo** [comment]
+1

**Pull request merged**